server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/Decision.java

```diff
@@ -118,8 +118,10 @@ private static Single readSingleFrom(StreamInput in) throws IOException {
     enum Type implements Writeable {
         // ordered by positiveness; order matters for serialization and comparison
         NO,
-        THROTTLE,
         NOT_PREFERRED,
+        // Temporarily throttled is a better choice than choosing a not-preferred node,
+        // but NOT_PREFERRED and THROTTLED are generally not comparable.
+        THROTTLE,
```

Comment:
I don't _think_ we can do this can we? It'll mean in the presence of a `THROTTLE` and a `NOT_PREFERRED`, `canAllocate` will return `NOT_PREFERRED`, which I think is bad for non-desired-balance allocation, because there we should wait for a `THROTTLE` to become a yes before allocating to a `NOT_PREFERRED`?